includes/abstracts/class-wc-gateway-mondido-abstract.php

```diff
@@ -331,20 +335,22 @@ public function handle_transaction( $order, $transaction_data ) {
 
 		// Check transaction was processed
 		$current_transaction_id = $order->get_transaction_id();
-		$current_status = get_post_meta( $order_id, '_mondido_transaction_status', true );
+		$current_status = $this->orderStorage->get_meta( $order, '_mondido_transaction_status', true );
+
 		if ( $current_transaction_id === $transaction_id && $current_status === $status ) {
 			throw new \Exception( "Transaction already applied. Order ID: {$order_id}. Transaction ID: {$transaction_id}. Transaction status: {$status}" );
 		}
 
-		// Save Transaction
-		delete_post_meta( $order_id, '_transaction_id' );
-		update_post_meta( $order_id, '_transaction_id', $transaction_id );
+		$this->orderStorage->delete_meta_data( $order, '_transaction_id' );
```

Comment:
This pair of lines with delete + update is repeating itself in multiple places. In order to reduce duplication we could introduce a `replace_meta_data` method on the order storage.